src/renderer/components/swap/Swap.utils.test.ts

```diff
@@ -293,4 +298,241 @@ describe('components/swap/utils', () => {
       )
     })
   })
+
+  describe('calcRefundFee', () => {
+    it('should be 3 x inbound fee', () => {
+      const result = calcRefundFee(baseAmount(2))
+      expect(eqBaseAmount.equals(result, baseAmount(6))).toBeTruthy()
+    })
+  })
+
+  describe('minBalanceToSwap', () => {
+    it('returns min. amount to cover inbound + refund fees', () => {
+      const params = {
+        inFee: {
+          amount: baseAmount(100),
+          asset: AssetBNB
+        }
+      }
+      const result = minBalanceToSwap(params)
+      expect(eqBaseAmount.equals(result, baseAmount(600))).toBeTruthy()
+    })
+  })
+
+  describe('minAmountToSwapMax1e8', () => {
+    const poolsData = {
+      'BNB.BUSD-74E': {
+        assetBalance: assetToBase(assetAmount(20)), // 1 BUSD = 0.05 RUNE
+        runeBalance: assetToBase(assetAmount(1)) // 1 RUNE = 20 BUSD
+      },
+      'ETH.USDT-0xa3910454bf2cb59b8b3a401589a3bacc5ca42306': {
+        assetBalance: assetToBase(assetAmount(20)), // 1 USDT = 0.05 RUNE
+        runeBalance: assetToBase(assetAmount(1)) // 1 RUNE = 20 USDT
+      },
+      'BNB.BNB': {
+        assetBalance: assetToBase(assetAmount(1)), // 1 BNB = 30 RUNE (600 USD)
+        runeBalance: assetToBase(assetAmount(30)) // 1 RUNE = 0.03 BNB
+      },
+      'ETH.ETH': {
+        assetBalance: assetToBase(assetAmount(1)), // 1 ETH = 100 RUNE (2000 USD)
+        runeBalance: assetToBase(assetAmount(100)) // 1 RUNE = 0.01 ETH
+      }
+    }
+
+    it('non chain asset -> chain asset (same chain): BNB.USD -> BNB.BNB)', () => {
+      const inAssetDecimal = 6
```

Comment:
maybe just BNB_DECIMAL ?

Reply:
_I think it's ok to add such "mock" value for testing..._